lib/install/client/automatic_lifecycle.go

```diff
@@ -52,24 +53,30 @@ func (r *AutomaticLifecycle) HandleStatus(ctx context.Context, c *Client, status
 		if err := r.generateDebugReport(ctx, c); err != nil {
 			c.WithError(err).Warn("Failed to generate debug report.")
 		}
-		if err := c.shutdown(ctx); err != nil && !isServerUnavailableError(err) {
+		var err error
+		if pb.IsFailedPreconditionError(statusErr) {
+			err = c.shutdownWithExitCode(ctx, defaults.FailedPreconditionExitCode)
```

Comment:
If the client was connected, it will make the service exit with the specific exit code. Otherwise, the service is supposed to exit with the same error code on its own.
Having client control the server being the preferred way allowed to simplify a lot of workflows previously, so this is sticking to it.